# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmBuiltins.cpp
# Commit: cf96078694ae
# Full Hash: cf96078694aea74cae4557f683a76c6de6ba9cf1
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2023-08-05 09:19:01
# Regressor Bug: 1692065
# File Overlap Count: 3
# Description:
#   Bug 1692065 - wasm: Rename MIRType::RefOrNull to 'WasmAnyRef'. r=yury
#   
#   This is a mechanical change. The previous name 'RefOrNull' was too
#   generic, and it wasn't clear that the type was actually a wasm::AnyRef
#   type.
# ==============================================================================

diff -r 662cb63c42d7 -r cf96078694ae js/src/wasm/WasmBuiltins.cpp
--- a/js/src/wasm/WasmBuiltins.cpp	Fri Aug 04 14:06:49 2023 +0000
+++ b/js/src/wasm/WasmBuiltins.cpp	Fri Aug 04 14:06:49 2023 +0000
@@ -69,7 +69,7 @@
 #define _I32 MIRType::Int32
 #define _I64 MIRType::Int64
 #define _PTR MIRType::Pointer
-#define _RoN MIRType::RefOrNull
+#define _RoN MIRType::WasmAnyRef
 #define _VOID MIRType::None
 #define _END MIRType::None
 #define _Infallible FailureMode::Infallible
@@ -425,7 +425,7 @@
     case MIRType::Int64:
       return ArgType_Int64;
     case MIRType::Pointer:
-    case MIRType::RefOrNull:
+    case MIRType::WasmAnyRef:
       return ArgType_General;
     case MIRType::Float32:
       return ArgType_Float32;